datamade/committee-oversight
# Production deployment

## Overview

Closes #123. After some certbot trials and an assist from @jeancochrane, this is working! Behold [committeeoversight-production.datamade.us](https://committeeoversight-production.datamade.us/). I also manually deployed this to staging to make sure it passed.

To transfer the database on [staging](committeeoversight.datamade.us), I first grabbed the last entry from the backups [S3 bucket](https://s3.console.aws.amazon.com/s3/buckets/datamade-postgresql-backups/hearings/?region=us-east-1&tab=overview) and saved it to my desktop as `hearings.dump`. I opened terminal and ran the following:

```
scp ~/Desktop/hearings.dump ubuntu@committeeoversight-production.datamade.us:/tmp
ssh ubuntu@committeeoversight-production.datamade.us
sudo su postgres
createdb hearings
pg_restore -C -j4 --no-owner -U postgres -d hearings /tmp/hearings.dump | psql
```

@hancush could we chat about how to handle two different sites re: regular scraping and backups?

A: Do a scrape on both sites at different times (ie, midnight & noon). Only back up from production!

## Testing Instructions

 * Visit the [staging](https://committeeoversight.datamade.us/) and [production](https://committeeoversight-production.datamade.us/) sites and make sure they're running as you'd expect
